Rubberduck.Core/UI/Command/Refactorings/CodePaneRefactorMoveContainingFolderCommand.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Refactorings.MoveFolder;
+using Rubberduck.UI.Command.Refactorings.Notifiers;
+using Rubberduck.VBEditor.Utility;
+
+namespace Rubberduck.UI.Command.Refactorings
+{
+    public class CodePaneRefactorMoveContainingFolderCommand : RefactorCodePaneCommandBase
+    {
+        private readonly RubberduckParserState _state;
+        private readonly ISelectedDeclarationProvider _selectedDeclarationProvider;
+
+        public CodePaneRefactorMoveContainingFolderCommand(
+            MoveContainingFolderRefactoring refactoring,
+            MoveContainingFolderRefactoringFailedNotifier failureNotifier,
+            ISelectionProvider selectionProvider,
+            RubberduckParserState state,
+            ISelectedDeclarationProvider selectedDeclarationProvider)
+            : base(refactoring, failureNotifier, selectionProvider, state)
+        {
+            _selectedDeclarationProvider = selectedDeclarationProvider;
+            _state = state;
+
+            AddToCanExecuteEvaluation(SpecializedEvaluateCanExecute);
+        }
+
+        private bool SpecializedEvaluateCanExecute(object parameter)
+        {
+            var target = GetTarget();
+
+            return target != null
+                   && target is ModuleDeclaration
+                   && !_state.IsNewOrModified(target.QualifiedModuleName);
```

Comment:
Wouldn't the null-check be redundant with `target is ModuleDeclaration module`?

e.g.

    return target is ModuleDeclaration module && !_state.IsNewOrModified(module.QualifiedModuleName);

Reply:
Actually the null-check is redundant even without a `module` variable.

Reply:
I will remove the `null` check.